src/services/admin.rs

```diff
@@ -84,4 +101,58 @@ mod tests {
 
         assert_eq!(restaurant_vec.len(), 0);
     }
+
+    #[actix_rt::test]
+    async fn test_extract_id() {
+        let db_pool = database::new_pool();
+
+        let mut app = init_service(App::new()
+        .data(db_pool.clone())
+        .service(super::load_data))
+        .await;
+
+
+        let mut restaurant_vec = Restaurant::get_restaurant_references(&db_pool.get().expect("Cant get database connection"));
+        assert_eq!(restaurant_vec.len(), 0);
+
+        //Send a request with a single restaurant in the body in json format
+        TestRequest::get()
+            .uri("/admin/insert")
+            .peer_addr(SocketAddr::new(IpAddr::V4(Ipv4Addr::new(54, 231, 65, 23)), 8080))
+            .method(Method::POST)
+            .set_payload("[{\"cvrnr\": \"27560946\",\"pnr\": \"1012180825\",\"region\": null,\"industry_code\": \"472400\",\"industry_text\": \"Detailhandel med brød, konditori- og sukkervarer\",\"start_date\": \"2005-06-01T00:00:00Z\",        \"smiley_reports\": [            {                \"report_id\": \"Virk1873390\",                \"smiley\": 1,                \"date\": \"2021-03-22T00:00:00Z\"            },            {                \"report_id\": \"Virk1834536\",                \"smiley\": 1,                \"date\": \"2020-11-11T00:00:00Z\"            },            {                \"report_id\": \"Virk1789610\",                \"smiley\": 1,                \"date\": \"2020-08-04T00:00:00Z\"            },            {                \"report_id\": \"Virk1771244\",                \"smiley\": 2,                \"date\": \"2020-06-10T00:00:00Z\"            }        ],        \"city\": \"Charlottenlund\",        \"elite_smiley\": \"0\",        \"geo_lat\": 55.762464,        \"geo_lng\": 12.585801,        \"franchise_name\": null,        \"niche_industry\": \"Bagere og bagerafdelinger\",        \"url\": \"http://www.findsmiley.dk/da-DK/Searching/DetailsView.htm?virk=757164\",        \"address\": \"Ordrup Jagtvej 42B, st\",        \"name\": \"Patricks Bake Shop - Ordrup ApS\",        \"name_seq_nr\": \"757164\",        \"zip_code\": \"2920\",        \"ad_protection\": \"0\",        \"company_type\": \"Detail\"    }]".as_bytes())
+            .send_request(&mut app)
+            .await;
+
+        restaurant_vec = Restaurant::get_restaurant_references(&db_pool.get().expect("Cant get database connection"));
+
+        assert_ne!(restaurant_vec.len(), 1);
```

Comment:
Shouldnt this assert be assert_eq? Because we're inserting one entry and thus expect to get 1 restaurant from the call